src/abTestingInfra/ABTestingContext/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { AB_TEST_PAYLOAD_PREFIX } from '../utils';
+import { useRouter } from 'next/router';
+import { ABTestingPayload } from '../types';
+
+const defaultABTestingContext: ABTestingPayload = {};
+const ABTestingContext = React.createContext(defaultABTestingContext);
+
+function getABTestContext(payload: string): ABTestingPayload {
+  const searchParams = new URLSearchParams(payload);
+
+  const result: ABTestingPayload = {};
+
+  // can't call reduce on searchParams, that's why using forEach
+  searchParams.forEach((value, key) => {
+    result[key] = value;
+  });
+
+  return result;
+}
+
+function getABTestContextFromPath(path: string): ABTestingPayload {
```

Comment:
An average reader may not know what `ABTestContext` lives in the path. What else could this be named?

Reply:
`extractABTestingContext`?

Reply:
Idk, that's pretty similar. Rather than saying _how_ the function gets the job done, what language can we use to define _what_ job is getting done? Like, does `parseVariantsToPresent` make sense?

Reply:
This function returns ABTestingPayload, so maybe `deserializeABTestingPayload`?  As form `parseVariantsToPresent` is too abstract